src/main/java/br/com/jackson/stop/jogo/JogoController.java

```diff
@@ -0,0 +1,81 @@
+package br.com.jackson.stop.jogo;
+
+import br.com.jackson.stop.compartilhado.anotacoes.ICP;
+import br.com.jackson.stop.sala.Sala;
+import br.com.jackson.stop.sala.SalaRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.Assert;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@RestController
+@RequestMapping("/api/jogo")
+@ICP(7)
+public class JogoController {
+
+  // 1
+  private final SalaRepository salaRepository;
+
+  @Autowired
+  public JogoController(SalaRepository salaRepository) {
+    this.salaRepository = salaRepository;
+  }
+
+  // fiquei na dúvida se é Post ou Get, ao mesmo tempo que buscamos os dados da sala, criamos um
+  // usuário e temos um json no corpo
+  @GetMapping
+  @Transactional
+  public DetalhesDaSalaEmJogoResponse jogoAleatorio( // 1
+      @Valid @RequestBody EntrarNoJogoRequest request) {
+    var salaDisponivel =
+        salaRepository.findAll().stream()
+            .filter(sala -> sala.salaComVagaDisponivel() && !sala.isPrivada())
```

Comment:
Se você faz uma condição onde os dois lados usam informação do mesmo objeto é pq deveria ter um método no objeto. 

Outro detalhe, para este projeto adiciona função como argumetno como item da métrica do cdd e trabalha com o limite de 7 pontos, tudo bem? Isso vai te forçar a dividir mais o código e treinar suas habilidades de modelagem.

Reply:
alterei o método, falta mudar o limite do CDD.